core/Archive/ArchiveInvalidator.php

```diff
@@ -276,14 +277,17 @@ public function markArchivesAsInvalidated(
         array $dates,
         $period,
         Segment $segment = null,
-        $cascadeDown = false,
-        $forceInvalidateNonexistentRanges = false,
-        $name = null,
-        $ignorePurgeLogDataDate = false
+        bool $cascadeDown = false,
+        bool $forceInvalidateNonexistentRanges = false,
+        ?string $name = null,
+        bool $ignorePurgeLogDataDate = false,
+        bool $doNotCreateInvalidations = false
     ) {
         $plugin = null;
         if ($name && strpos($name, '.') !== false) {
             list($plugin) = explode('.', $name);
+        } elseif ($name) {
+            $plugin = $name;
```

Comment:
This is a small fix to the plugin detection. When a name is provided, but doesn't contain a `.`, it is actually the plugin name only. So we should check that correctly.